prover/zkevm/witness.go

```diff
@@ -16,21 +17,33 @@ type Witness struct {
 	// proof trace generation.
 	ExecTracesFPath string
 	// StateManager traces
-	SMTraces        [][]statemanager.DecodedTrace
-	TxSignatures    map[[32]byte]ethereum.Signature
+	SMTraces [][]statemanager.DecodedTrace
+	// TxSignatures lists the signatures of the transaction as found
+	// chronologically in the block.
+	TxSignatures []ethereum.Signature
+	// TxHashes lists the hash of the transactions in the order found in the
+	// block.
+	TxHashes        [][32]byte
 	L2BridgeAddress common.Address
 	ChainID         uint
 }
 
-func (w Witness) TxSignatureGetter(txHash []byte) (r, s, v *big.Int, err error) {
-	var (
-		sig, found = w.TxSignatures[[32]byte(txHash)]
-	)
+// TxSignatureGetter implements the ecdsa.TxSignatureGetter interface
+func (w Witness) TxSignatureGetter(i int, txHash []byte) (r, s, v *big.Int, err error) {
 
-	if !found {
-		return nil, nil, nil, fmt.Errorf("could not find signature for tx hash = 0x%x", txHash)
+	if i > len(w.TxHashes) {
+		return nil, nil, nil, fmt.Errorf("requested txID outgoes the total number of transactions we found in the conflation")
 	}
 
+	if utils.HexEncodeToString(txHash) != utils.HexEncodeToString(w.TxHashes[i][:]) {
+		return nil, nil, nil, fmt.Errorf(
+			"requested txID=%v while txnrlp expects it to have it for txhash=%v but the blocks transaction has hash=%v",
```

Comment:
There are some existing terms related to a block.
`transactionIndex`: When a transaction is added to a block, it is assigned a numbered position within that block. The transactionIndex returns the position of the transaction in the block it has been added to.
(https://docs.alchemy.com/docs/understanding-the-transaction-object-on-ethereum#transactionindex)

It is referred to as `position in block` on Etherscan.

Here `txID` is similar but within a conflation, correct?

Reply:
Yes correct, that would be position in conflation